### PR TITLE
Add network create flags --scope, --config-only, --config-from

### PR DESCRIPTION
**Pull request by aboch**
This is for moby/moby/pull/32981 and depends on the moby PR to be merged first.

### CODE REVIEW

**Comment by thaJeztah** [comment]
Looks like this needs a rebase :cry:

**Comment by cpuguy83** [comment]
Let's keep this in design review until the API side is merged.

File: cli/command/network/create.go [hunk 1]
```diff
@@ -96,7 +105,15 @@ func runCreate(dockerCli *command.DockerCli, options createOptions) error {
 		EnableIPv6:     options.ipv6,
 		Attachable:     options.attachable,
 		Ingress:        options.ingress,
-		Labels:         runconfigopts.ConvertKVStringsToMap(options.labels.GetAll()),
+		Scope:          options.scope,
+		ConfigOnly:     options.configOnly,
+		Labels:         runconfigopts.ConvertKVStringsToMap(opts.labels.GetAll()),
```
Comment (mavenugo) [inline cli/command/network/create.go]: shouldnt this be `options.labels` ?

File: cli/command/network/create.go [hunk 2]
```diff
@@ -96,7 +105,15 @@ func runCreate(dockerCli *command.DockerCli, options createOptions) error {
...
+	}
+
+	if from := options.configFrom; from != "" {
+		nc.ConfigFrom = network.ConfigReference{
```
Comment (mavenugo) [inline cli/command/network/create.go]: &network.ConfigReference

File: cli/command/network/create.go (hunk 1 above)
Comment (aboch) [inline cli/command/network/create.go]: thanks, leftover form merge conflict

**Comment by aboch** [comment]
Thanks @thaJeztah @mavenugo Updated

**Comment by nishanttotla** [comment]
@aboch @mavenugo please include https://github.com/moby/moby/pull/33239 in the vendor here.

**Comment by aboch** [comment]
@nishanttotla moby/moby#33239 is not changing any docker pkg exported function.
Even after it is being merged, I am not sure any of those changes would be imported from a vendoring to here.

**Comment by nishanttotla** [comment]
@aboch it is changing the `client` package. That part is vendored into `docker/cli` and required here.

**Comment by aboch** [comment]
Oh I see, the whole client pkg is being used here. Got it.

**Comment by mavenugo** [comment]
@nishanttotla we also have https://github.com/docker/cli/pull/62 which will be even later.
So, I dont see a reason to block this PR

**Comment by eyz** [comment]
> This will need a rebase after moby/moby#32981 is merged and vendor updated.

Time to rebase?

**Comment by mavenugo** [comment]
@eyz we are also waiting to pick up https://github.com/moby/moby/pull/33239 as per @nishanttotla request.

**Comment by aboch** [comment]
@mavenugo Updated vendoring to include moby/moby#33239 and CI is now green

**Pull request merged**